opcua/server/history.py

```diff
@@ -108,20 +117,191 @@ def read_node_history(self, node, start, end, nb_values):
                     results = results[-nb_values:]
             return results, cont
 
+    def new_historized_event(self, event, period):
+        self._events = []
+
     def save_event(self, event):
         raise NotImplementedError
 
     def read_event_history(self, start, end, evfilter):
         raise NotImplementedError
 
+    def stop(self):
+        pass
+
+
+class HistorySQLite(HistoryStorageInterface):
+    """
+    very minimal history backend storing data in SQLite database
+    """
+    # FIXME: need to check on if sql_conn.commit() should be inside try block; and if .rollback() needs to be except
+
+    def __init__(self):
+        self._datachanges_period = {}
+        self._events = {}
+        self._db_file = "history.db"
+
+        # SQL objects must be accessed in a single thread
+        # adding a new node to be historized probably happens on the main thread
+        self._conn_new = None
+        self._c_new = None
+
+        # subscriptions are in another thread so it needs it's own sqlite connection object
+        self._conn_sub = None
+        self._c_sub = None
+
+        # FIXME: no idea what thread the read values happen in, just make a new conn object for now
+        self._conn_read = None
+        self._c_read = None
```

Comment:
I have never used locks and I don't know how they work. Even with a lock I think the SQL module will complain. Anyways I will try to do this.